perses/app/fah_generator.py

```diff
@@ -0,0 +1,203 @@
+from perses.dispersed.feptasks import minimize
+from openmmtools import SamplerState, ThermodynamicState, CompoundThermodynamicState
+
+from openmmtools import mcmc, utils
+from perses.utils.smallmolecules import render_atom_mapping
+from perses.tests.utils import validate_endstate_energies
+from perses.dispersed.smc import SequentialMonteCarlo
+from simtk.openmm import XmlSerializer
+from perses.utils import data
+from openmmtools.integrators import PeriodicNonequilibriumIntegrator
+from copy import deepcopy
+import tqdm
+from openmmtools.integrators import LangevinIntegrator
+
+import datetime
+class TimeFilter(logging.Filter):
+    def filter(self, record):
+        try:
+          last = self.last
+        except AttributeError:
+          last = record.relativeCreated
+        delta = datetime.datetime.fromtimestamp(record.relativeCreated/1000.0) - datetime.datetime.fromtimestamp(last/1000.0)
+        record.relative = '{0:.2f}'.format(delta.seconds + delta.microseconds/1000000.0)
+        self.last = record.relativeCreated
+        return True
+
+fmt = logging.Formatter(fmt="%(asctime)s:(%(relative)ss):%(name)s:%(message)s")
+#logging.basicConfig(level = logging.NOTSET)
+logging.basicConfig(
+    format='%(asctime)s %(levelname)-8s %(message)s',
+    level=logging.INFO,
+    datefmt='%Y-%m-%d %H:%M:%S')
+_logger = logging.getLogger()
+_logger.setLevel(logging.INFO)
+[hndl.addFilter(TimeFilter()) for hndl in _logger.handlers]
+[hndl.setFormatter(fmt) for hndl in _logger.handlers]
+
+setup_yaml = sys.argv[1]
+neq_yaml = sys.argv[2]
+ENERGY_THRESHOLD=1e-4
+x = 'lambda'
+DEFAULT_ALCHEMICAL_FUNCTIONS = {
+                             'lambda_sterics_core': x,
+                             'lambda_electrostatics_core': x,
+                             'lambda_sterics_insert': f"select(step({x} - 0.5), 1.0, 2.0 * {x})",
+                             'lambda_sterics_delete': f"select(step({x} - 0.5), 2.0 * ({x} - 0.5), 0.0)",
+                             'lambda_electrostatics_insert': f"select(step({x} - 0.5), 2.0 * ({x} - 0.5), 0.0)",
+                             'lambda_electrostatics_delete': f"select(step({x} - 0.5), 1.0, 2.0 * {x})",
+                             'lambda_bonds': x,
+                             'lambda_angles': x,
+                             'lambda_torsions': x}
+
+from openmmtools.constants import kB
+
+def make_neq_integrator(setup_options_dict, neq_options_dict, alchemical_functions = DEFAULT_ALCHEMICAL_FUNCTIONS):
+    integrator_kwargs = deepcopy(setup_options_dict)
+    integrator_kwargs.update(neq_options_dict)
+    integrator_kwargs['alchemical_functions'] = alchemical_functions
+    integrator = PeriodicNonequilibriumIntegrator(**integrator_kwargs)
+    return integrator
+
+def relax_structure(temperature, system, positions, nminimize=100, nequil = 4, n_steps_per_iteration=250):
+    integrator = LangevinIntegrator(temperature = temperature)
+    context = openmm.Context(system, integrator)
```

Comment:
Are we guaranteed that we don't need to explicitly set the context parameters to anything but their default values to have hybrid state A? I was concerned this might be necessary.

Reply:
yes